CosNet.AppUI.Android/app/src/main/java/cosnet/android/EditCosplay.java

```diff
@@ -0,0 +1,141 @@
+package cosnet.android;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+
+import android.app.AlertDialog;
+import android.app.DatePickerDialog;
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageButton;
+import android.widget.Spinner;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+
+import cosnet.android.Entities.Cosplay;
+import me.abhinay.input.CurrencyEditText;
+
+public class EditCosplay extends AppCompatActivity {
+
+  private static final String TAG = "EditCosplay";
+
+  private CosnetDb db;
+
+  private Cosplay cosplay;
+  private EditText characterEditText;
+  private EditText seriesEditText;
+  private EditText startDateEditText;
+  private EditText dueDateEditText;
+  private CurrencyEditText budgetEditText;
+  private Spinner statusSpinner;
+  private Button saveButton;
+  private List<String> statusses;
+  private int year;
+  private int month;
+  private int day;
+
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_edit_cosplay);
+
+    Toolbar toolbar = findViewById(R.id.toolbar);
+    setSupportActionBar(toolbar);
+    getSupportActionBar().setDisplayShowTitleEnabled(false);
+    getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+    //get cosplay from intent
+    Intent incomingIntent = getIntent();
+    cosplay = (Cosplay) incomingIntent.getSerializableExtra("cosplay");
+
+    db = CosnetDb.getInstance(this);
+
+    characterEditText = findViewById(R.id.characterEditText);
```

Comment:
This block (62-80) should be extracted into a separate method for better readability.